kroxylicious/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -246,6 +107,9 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
             }
         }
         else {
+            if (!(msg instanceof OpaqueResponseFrame)) {
+                LOGGER.warn("Unexpected message reading from upstream: {}", msg, new RuntimeException());
```

Comment:
That looks odd, why raising that rather meaningless exception?

Reply:
Ah, that was a hangover from me debugging my code, wanting to see in the stack what might be responsible for these other `msg` types, and leading to the discovery of #61. Does `IllegalStateException` suit you better?